lib/ansible/modules/database/postgresql/postgresql_set.py

```diff
@@ -0,0 +1,518 @@
+version_added: "2.8"
+options:
+  name:
+    description:
+    - Name of PostgreSQL server parameter.
+    type: str
+    required: true
+  value:
+    description:
+    - Parameter value to set.
+    - To remove parameter string from postgresql.auto.conf and
+      reload the server configuration you must pass I(value=default).
+      With I(value=default) the playbook always returns changed is true.
+    type: str
+    required: true
+  reset:
+    description:
+    - Restore parameter to initial state (boot_val). Mutually exclusive with I(value).
+    type: bool
+    default: false
+  session_role:
+    description:
+    - Switch to session_role after connecting. The specified session_role must
+      be a role that the current login_user is a member of.
+    - Permissions checking for SQL commands is carried out as though
+      the session_role were the one that had logged in originally.
+  db:
+    description:
+    - Name of database to connect.
+    type: str
+    aliases:
+    - login_db
+  port:
+    description:
+    - Database port to connect.
+    type: int
+    default: 5432
+    aliases:
+    - login_port
+  login_user:
+    description:
+    - User (role) used to authenticate with PostgreSQL.
+    type: str
+    default: postgres
+  login_password:
+    description:
+    - Password used to authenticate with PostgreSQL.
+    type: str
+  login_host:
+    description:
+    - Host running PostgreSQL.
+    type: str
+  login_unix_socket:
+    description:
+    - Path to a Unix domain socket for local connections.
+    type: str
+  ssl_mode:
+    description:
+    - Determines whether or with what priority a secure SSL TCP/IP connection
+      will be negotiated with the server.
+    - See U(https://www.postgresql.org/docs/current/static/libpq-ssl.html) for
+      more information on the modes.
+    - Default of C(prefer) matches libpq default.
+    type: str
+    choices: [ allow, disable, prefer, require, verify-ca, verify-full ]
+    default: prefer
+  ssl_rootcert:
+    description:
+    - Specifies the name of a file containing SSL certificate authority (CA)
+      certificate(s).
+    - If the file exists, the server's certificate will be
+      verified to be signed by one of these authorities.
+    type: str
+notes:
+- Supported version of PostgreSQL is 9.4 and later.
+- Pay attention, change setting with 'postmaster' context can return changed is true
+  when actually nothing changes because the same value may be presented in
+  several different form, for example, 1024MB, 1GB, etc. However in pg_settings
+  system view it can be defined like 131072 number of 8kB pages.
+  The final check of the parameter value cannot compare it because the server was
+  not restarted and the value in pg_settings is not updated yet.
+- For some parameters restart of PostgreSQL server is required.
+  See official documentation U(https://www.postgresql.org/).
```

Comment:
too general link. Would be better to use for example https://www.postgresql.org/docs/current/view-pg-settings.html where all possible contexts values are described

Reply:
good job